File: objc/Builder/Builder/GamingComputerBuilder.m
#import "GamingComputerBuilder.h"

@implementation GamingComputerBuilder

-(void)buildCPU {
    [self.computer setCpu:@"Intel Core i7"];
}

-(void)buildMotherboard {
    [self.computer setMotherboardModel:@"16 GB"];
}

-(void)buildRAM {
    [self.computer setMotherboardModel:@"Gigabyte GA-Z170X"];
}

-(void)buildStorage {
    [self.computer setStorageValue:@"1 TB"];
}

-(void)buildGraphicsCard {
    [self.computer setGraphicsCardModel:@"Nvidia 1070"];
}

@end

File: swift/Builder/Builder/GamingComputerBuilder.swift
import Foundation

public class GamingComputerBuilder : ComputerBuilder {
    override func buildCPU() {
        self.computer?.cpu = "Intel Core i7"
    }
    
    override func buildMotherboard() {
        self.computer?.motherboardModel = "Gigabyte GA-Z170X"
    }
    
    override func buildRAM() {
        self.computer?.ramValue = "16 GB"
    }
    
    override func buildStorage() {
        self.computer?.storageValue = "1 TB"
    }
    
    override func buildGraphicsCard() {
        self.computer?.graphicsCardModel = "Nvidia 1070"
    }
}
